### PR TITLE
feat: use shiki for docs code blocks

### PR DESCRIPTION
**Pull request by dstaley**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-dshds-codeblock-docs-hashicorp.vercel.app/) 🔎

## 🗒️ What

This PR updates the code blocks used in documentation from our old `react-code-block` component to our new HDS `CodeBlock` component. In conjunction with this, we also switch from using Prism to Shiki for syntax highlighting.

## 🤷 Why

The new HDS CodeBlock is specifically designed to have great accessibility, and to fit in with the overall design of HashiCorp Developer.

## 🛠️ How

There are two main components to this PR.

1. Refactor of `mdx-components/mdx-code-blocks` to use `CodeBlock`. This required modifications to the `MdxCodeBlockConfig`, `MdxCodeTabs`, and `MdxPre` components to render HDS CodeBlocks instead of `react-code-block`.
2. Replacing our `rehypePrism` plugin with `rehypeShiki`, which allows us to correctly syntax highlight HCL code.

## 🧪 Testing

This PR is expansive, and impacts every page containing a code block. Since there are significant visual differences, visual regression testing is not possible. To test, navigate to a variety of documentation pages and ensure that features such as syntax highlighting, code tabs, copy buttons, and headers work in a similar fashion to production. To help make that easier, here are docs using each of those features:

- [Syntax highlighting](https://dev-portal-git-dshds-codeblock-docs-hashicorp.vercel.app/terraform/language/data-sources)
- [Code Tabs](https://dev-portal-git-dshds-codeblock-docs-hashicorp.vercel.app/terraform/cdktf/concepts/constructs)
- [Copy Buttons](https://dev-portal-git-dshds-codeblock-docs-hashicorp.vercel.app/terraform/language/modules/syntax)
- [Headers](https://dev-portal-git-dshds-codeblock-docs-hashicorp.vercel.app/vault/docs/enterprise/sentinel)

### CODE REVIEW

File: src/components/dev-dot-content/mdx-components/mdx-code-blocks/index.tsx [hunk 1]
```diff
@@ -3,23 +3,61 @@
...
+	heading?: string
+	wrapCode?: boolean
+}
+
+interface CodeBlockProps {
+	className?: string
+	code: ReactNode
+	language?: string
+	theme?: 'light' | 'dark'
+	hasBarAbove?: boolean
```
Comment (heatlikeheatwave) [inline src/components/dev-dot-content/mdx-components/mdx-code-blocks/index.tsx:30]: What does `hasBarAbove` control?

File: src/components/dev-dot-content/mdx-components/mdx-code-blocks/index.tsx [hunk 2]
```diff
@@ -29,12 +67,28 @@ export const MdxCodeBlockConfig = CodeBlockConfig
...
+						: `language-${restprops.language}`
+				const slugFromClass = languageClass.split('-')[1]
+				const heading = getLanguageName(slugFromClass) ?? slugFromClass
+
+				return (
+					<Tab key={slugFromClass} heading={heading} group={slugFromClass}>
+						{child}
+					</Tab>
+				)
+			})}
```
Comment (heatlikeheatwave) [inline src/components/dev-dot-content/mdx-components/mdx-code-blocks/index.tsx:90]: What do you think about splitting out this logic? Something like this (I've also taken a liberty of refactoring the `languageClass` line for clarity)

```
const setTab = (child) => {
  if (!(typeof child === 'object' && 'props' in child)) return null;

  const { mdxType, ...restprops } = child.props;
  const isCodeBlock = ['pre', 'CodeBlockConfig'].includes(mdxType);
  const languageClass = isCodeBlock ? restprops.children.props.className : `language-${restprops.language}`;
  const slugFromClass = languageClass.split('-')[1];
  const heading = getLanguageName(slugFromClass) ?? slugFromClass;

  return (
    <Tab key={slugFromClass} heading={heading} group={slugFromClass}>
      {child}
    </Tab>
  );
};

return (
  <Tabs>
    {childCodeBlocks.map(setTab)}
  </Tabs>
);
```

File: src/components/dev-dot-content/mdx-components/mdx-code-blocks/index.tsx [hunk 3]
```diff
@@ -29,12 +67,28 @@ export const MdxCodeBlockConfig = CodeBlockConfig
...
+		</Tabs>
```
Comment (heatlikeheatwave) [inline src/components/dev-dot-content/mdx-components/mdx-code-blocks/index.tsx:91]: What do you think about splitting out this logic? Something like this (I've also taken a liberty of refactoring the `languageClass` line for clarity)

```
const setTab = (child) => {
  if (!(typeof child === 'object' && 'props' in child)) return null;

  const { mdxType, ...restprops } = child.props;
  const isCodeBlock = ['pre', 'CodeBlockConfig'].includes(mdxType);
  const languageClass = isCodeBlock ? restprops.children.props.className : `language-${restprops.language}`;
  const slugFromClass = languageClass.split('-')[1];
  const heading = getLanguageName(slugFromClass) ?? slugFromClass;

  return (
    <Tab key={slugFromClass} heading={heading} group={slugFromClass}>
      {child}
    </Tab>
  );
};

return (
  <Tabs>
    {childCodeBlocks.map(setTab)}
  </Tabs>
);
```

File: src/components/dev-dot-content/mdx-components/mdx-code-blocks/language-names.json [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+	"less": "LESS",
+	"lilypond": "LilyPond",
+	"liquid": "Liquid",
+	"elisp": "Emacs Lisp",
+	"emacs": "Emacs",
+	"emacs-lisp": "Emacs Lisp",
+	"lisp": "Lisp",
+	"livescript": "LiveScript",
+	"llvm": "LLVM",
+	"lolcode": "LOLCODE",
```
Comment (nandereck) [inline src/components/dev-dot-content/mdx-components/mdx-code-blocks/language-names.json:137]: Hah what!

**Review by nandereck — APPROVED** [review]
Wow - this is really cool! LGTM!

**Pull request merged**